### PR TITLE
[FIX] sale, purchase: fix inconsistent invoicing threshold tests

### PR DESCRIPTION
**Pull request by ushyme**
The tests introduced by https://github.com/odoo/odoo/commit/6605cc6673b8d46e1978c788fb48fc3828841112 were found to be inconsistent.

This commit fixes that.

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
looks good to me, this is to prevent false positive that seems could happen sometimes with

https://github.com/odoo/odoo/commit/6605cc6673b8d46e1978c788fb48fc3828841112

but I could not reproduce the issue locally with or without the test :thinking: 

ps: maybe better to have "fix inconsistent thresold test" in title so it's more greppable

File: addons/purchase/tests/test_purchase_invoice.py [hunk 1]
```diff
@@ -502,6 +502,8 @@ def test_partial_billing_interaction_with_invoicing_switch_threshold(self):
 
         self.assertEqual(line.qty_invoiced, 10)
 
+        invoice.invalidate_cache(fnames=['payment_state', 'state'])
```
Comment (nle-odoo) [inline addons/purchase/tests/test_purchase_invoice.py:505]: shouldn't this be below the invoicing_switch_threshold? if I understood this was what was changing the value so it makes more sense functionally (even if it should work if we don't access the field values inbetween)

Comment (ushyme) [reply addons/purchase/tests/test_purchase_invoice.py:505]: Oops, you're right, it should be below the invoicing_switch_threshold.
I'll fix it, thanks !

**Review by nle-odoo — APPROVED** [review]
looks good to me

this doesn't seem to happen much in runbot (it happened at merge time), I'm not sure why (maybe a recompute in another installed module that doesn't happen in local) but I can reproduce easily with just sale, purchase, account_accountant

robodoo r+

**Pull request closed without merging**